File: shoebill_libs/LtrpApi/src/main/java/lt/ltrp/data/LtrpWeaponData.java
package lt.ltrp.data;

import lt.ltrp.object.Entity;
import net.gtaun.shoebill.constant.WeaponModel;
import net.gtaun.shoebill.data.Vector2D;
import net.gtaun.shoebill.data.WeaponData;

/**
 * @author Bebras
 *         2015.12.02.
 */
public class LtrpWeaponData extends WeaponData implements Entity {



    private int uuid;
    public boolean job;
    private boolean isDropped;


    public LtrpWeaponData(WeaponModel type, int ammo, boolean job) {
        super(type, ammo);
        this.job = job;
    }

    public LtrpWeaponData(int uuid, WeaponModel type, int ammo, boolean job) {
        super(type, ammo);
        this.job = job;
        this.uuid = uuid;
    }

    public LtrpWeaponData() {

    }


    public LtrpWeaponData(boolean job) {
        this.job = job;
    }

    public LtrpWeaponData(WeaponData data, boolean job) {
        super(data);
        this.job = job;
    }

    @Override
    public LtrpWeaponData clone() {
        return new LtrpWeaponData(this.getModel(), this.getAmmo(), this.isJob());
    }

    public boolean isJob() {
        return job;
    }

    public void setJob(boolean job) {
        this.job = job;
    }

    public boolean isDropped() {
        return isDropped;
    }

    public void setDropped(Vector2D loc) {
        this.isDropped = true;
        new DroppedWeaponData(this, loc);
    }

    @Override
    public void setUUID(int uuid) {
        this.uuid = uuid;
    }

    @Override
    public int getUUID() {
        return uuid;
    }
}

File: shoebill_libs/LtrpApi/src/main/java/lt/ltrp/data/LtrpWeaponData.kt
package lt.ltrp.data

import lt.ltrp.`object`.Entity
import lt.ltrp.`object`.impl.EntityImpl
import net.gtaun.shoebill.constant.WeaponModel
import net.gtaun.shoebill.data.Vector2D
import net.gtaun.shoebill.data.WeaponData

/**
 * @author Bebras
 * *         2015.12.02.
 */
open class LtrpWeaponData(weaponData: WeaponData, job: Boolean, uuid: Int) : EntityImpl(uuid) {

    var weaponData: WeaponData = weaponData;
    var isJob: Boolean = false
    var isDropped: Boolean = false
        private set


    constructor(type: WeaponModel, ammo: Int, job: Boolean) : this(WeaponData(type, ammo), job, Entity.INVALID_ID) {

    }

    constructor(uuid: Int, type: WeaponModel, ammo: Int, job: Boolean) : this(WeaponData(type, ammo), job, uuid) {

    }


    fun clone(): LtrpWeaponData {
        return LtrpWeaponData(this.weaponData.model, this.weaponData.ammo, this.isJob)
    }

    fun setDropped(loc: Vector2D) {
        this.isDropped = true
        DroppedWeaponData(this, loc)
    }


    companion object {
        val UNARMED = LtrpWeaponData(WeaponModel.UNKNOWN, 1, false)
    }
}
